ext/opentelemetry-ext-system-metrics/src/opentelemetry/ext/system_metrics/__init__.py

```diff
@@ -0,0 +1,165 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Instrument to report system (CPU, memory, network) and
+process (CPU, memory, garbage collection) metrics.
+
+Usage
+-----
+
+.. code:: python
+
+    from opentelemetry.ext.system_metrics import SystemMetrics
+    from opentelemetry.sdk.metrics.export import ConsoleMetricsExporter
+    from opentelemetry.sdk.metrics.export.controller import PushController
+
+    exporter = ConsoleMetricsExporter()
+    SystemMetrics(exporter)
+
+    # metrics are collected asynchrnously
+    input("...")
+
+API
+---
+"""
+
+import gc
+import os
+import typing
+
+import psutil
+
+from opentelemetry import metrics
+from opentelemetry.sdk.metrics.export import MetricsExporter
+from opentelemetry.sdk.metrics.export.controller import PushController
+
+_DEFAULT_INTERVAL = 30
+
+
+class SystemMetrics:
+    def __init__(
+        self,
+        exporter: MetricsExporter,
+        interval: int = _DEFAULT_INTERVAL,
+        labels: typing.Optional[typing.Dict[str, str]] = None,
+    ):
+        labels = {} if labels is None else labels
+        self.meter = metrics.get_meter(__name__)
+        self.controller = PushController(
+            meter=self.meter, exporter=exporter, interval=interval
+        )
+        self._proc = psutil.Process(os.getpid())
+        self._labels = labels
+
+        self.meter.register_observer(
+            callback=self._get_system_memory,
+            name="system.mem",
+            description="System memory",
+            unit="bytes",
+            value_type=int,
+            label_keys=self._labels.keys(),
+        )
+
+        self.meter.register_observer(
+            callback=self._get_system_cpu,
+            name="system.cpu",
+            description="System CPU",
+            unit="seconds",
+            value_type=float,
+            label_keys=self._labels.keys(),
+        )
+
+        self.meter.register_observer(
+            callback=self._get_network_bytes,
+            name="system.net.bytes",
+            description="System network bytes",
+            unit="bytes",
+            value_type=int,
+            label_keys=self._labels.keys(),
+        )
+
+        self.meter.register_observer(
+            callback=self._get_runtime_memory,
+            name="runtime.python.mem",
+            description="Runtime memory",
+            unit="bytes",
+            value_type=int,
+            label_keys=self._labels.keys(),
+        )
+
+        self.meter.register_observer(
+            callback=self._get_runtime_gc_count,
+            name="runtime.python.gc.count",
+            description="Runtime: gc objects",
+            unit="objects",
+            value_type=int,
+            label_keys=self._labels.keys(),
+        )
+
+    def _get_system_memory(self, observer: metrics.Observer) -> None:
+        """Observer callback for memory available
+        Args:
```

Comment:
Nit: A blank line is required before `Args` to render the documentation correctly.

Reply:
Thanks for catching this.